### PR TITLE
Issue 483

### PR DESCRIPTION
**Pull request by fvgh**
Proposed fix for  #483.
The changes remove duplication in the Spotless build scripts.
`_ext` is build with Gradle 6 like the rest of the Spotless projects.
The `_ext` projects become sub-projects of the Spotless root project, but are deactivated by default.
The `_ext` projects are activated by root project properties which allow activation by command line as well as per user `gradle.properties`.

The changes additionally contain:
* Work-around of WTP build (latest EFS changes broke the build, fix will be applied with next WTP update). Due to the dependency lock, the current WTP releases are not affected by the EFS change.
* Fix of JDT tests which are broken since #480

### CODE REVIEW

**Comment by fvgh** [comment]
@nedtwigg Please have a look at the mechanism to activate `_ext` projects. I changed the approach, since I did not wanted to fumble with the Eclipse start-up. I think the user properties is an adequate location to store such user specific modification to the build process.
The p2 conversion still is sometimes buggy (clean is required). But the clean&build for dedicated sub-projects works fine.

File: _ext/eclipse-wtp/gradle.properties [hunk 1]
```diff
@@ -19,8 +13,7 @@ VER_ECLISPE_EMF=[2.16.0,3.0.0[
 VER_ECLIPSE_OSGI_SERVICES=[3.8.0,4.0.0[
 VER_ECLIPSE_FILE_BUFFERS=[3.6.700,4.0.0[
 VER_ECLISPE_JFACE=[3.15.300,4.0.0[
-VER_ECLISPE_EFS=[1.7.500,2.0.0[
+#VER_ECLISPE_EFS=[1.7.500,2.0.0[
+#TODO: 1.7.600 changed bundle activation. Code adaptation required.
+VER_ECLISPE_EFS=1.7.500
```
Comment (nedtwigg) [inline _ext/eclipse-wtp/gradle.properties]: No comment needed, just noting the change.

File: _ext/gradle/java-setup.gradle [hunk 2]
```diff
@@ -1,78 +1,19 @@
...
+apply from: rootProject.file('gradle/java-setup.gradle')
 
 // setup java
 apply plugin: 'java'
 
-sourceCompatibility = ext_VER_JAVA
-targetCompatibility = ext_VER_JAVA
+// Show warning locations
 tasks.withType(JavaCompile) { 
-	options.encoding = 'UTF-8'
```
Comment (nedtwigg) [inline _ext/gradle/java-setup.gradle]: Why not `UTF-8`?

File: _ext/eclipse-wtp/gradle.properties (hunk 1 above)
Comment (fvgh) [reply _ext/eclipse-wtp/gradle.properties]: Brief comment :wink: 
Change allows check of this proposal by `./gradlew --write-locks -Pcom.diffplug.spotless.include_ext=true build`

File: _ext/gradle/java-setup.gradle (hunk 2 above)
Comment (fvgh) [reply _ext/gradle/java-setup.gradle]: Clean-up
UFT-8 required e.g. for:
* author names (as class comments)
* code comments in foreign language
* multi language UI

Think I once needed it when I refereed to the origin of "stolen" Eclipse code.
Due to service support / stubbing of `eclipse-base` such ugly work-arounds should not be required anymore.

**Comment by fvgh** [comment]
@nedtwigg  Please have a quick look at last comments and changes. I would prefer a squashed commit, since all changes are just related to build clean-up.

File: _ext/gradle/java-setup.gradle (hunk 2 above)
Comment (nedtwigg) [reply _ext/gradle/java-setup.gradle]: We have `UTF-8` enabled for the main projects: https://github.com/diffplug/spotless/blob/e870e78999887595d8480c699c4ef71c9e974b32/gradle/java-setup.gradle#L11

I believe it is needed due to string constants used in the diff formatter, if memory serves.  It's okay for `_ext` to be different, I'll let you decide.

**Comment by nedtwigg** [comment]
Looks great, I'll let you make a call re: UTF-8 above, then you can squash and merge :)

**Comment by fvgh** [comment]
@nedtwigg Sorry, was sleeping. Missed that the PR title is used as comment. Think the only way back is an amend and forced push, isn't it?

**Comment by nedtwigg** [comment]
Correct, but I think it's fine :)

**Pull request merged**